clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/model/KubernetesV2LoadBalancer.java

```diff
@@ -58,4 +62,57 @@ public static KubernetesV2LoadBalancer fromCacheData(
     }
     return new KubernetesV2LoadBalancer(manifest, cd.getId(), loadBalancerServerGroups);
   }
+
+  @Override
+  public String getType() {
+    return KubernetesCloudProvider.ID;
+  }
+
+  @Override
+  public String getName() {
+    return getManifest().getFullResourceName();
+  }
+
+  @Override
+  public String getDisplayName() {
+    return getManifest().getName();
+  }
+
+  @Override
+  public KubernetesApiVersion getApiVersion() {
+    return getManifest().getApiVersion();
+  }
+
+  @Override
+  public String getNamespace() {
+    return getManifest().getNamespace();
+  }
+
+  public String getRegion() {
```

Comment:
Why is namespace returned for both getNamespace and getRegion? Is the concept of region used by other providers but for Kubernetes it's equivalent to namespace?

Reply:
Yes, exactly, other providers have a concept of region and zone (where a server group would be in a region, and its instances in particular zones).  For the Kubernetes provider, we basically just return the namespace for both of those.

It's not really a perfect mapping, Kubernetes clusters do also have a concept of region/zone...as the instances in a cluster are a in a particular region/zone, and if you want high-availability you probably run your workloads in >1 cluster in >1 region/zone.  But the underlying infrastructure of a cluster is something that the Spinnaker Kubernetes provider has no knowledge about, so in the absence of that we just use namespace as region or zone.  In a lot of places where we're writing Kubernetes code it's clearer to just call it a namespace, so we return the namespace as `getNamespace` as well to avoid needing callers who know it's a Kubernetes object to need to ask for the region/zone.